pkg/controller/bastion/options.go

```diff
@@ -130,30 +130,25 @@ func ingressPermissions(bastion *extensionsv1alpha1.Bastion) ([]string, error) {
 
 		normalisedCIDR := ipNet.String()
 
-		if ip.To4() != nil {
-			cidrs = append(cidrs, normalisedCIDR)
-		} else if ip.To16() != nil {
-			// Only IPv4 is supported in sourceRanges[].
-			// https://cloud.google.com/compute/docs/reference/rest/v1/firewalls/insert
-			return nil, fmt.Errorf("IPv6 is currently not fully supported: %w", err)
+		if ip.To4() == nil {
+			return nil, fmt.Errorf("ip address is not valid: %w", err)
```

Comment:
fa2d1b9f

Reply:
```go
if ip.To4() == nil || ip.To16() == nil {
	return nil, fmt.Errorf("ip address is not valid: %w", err)
}
```
@tedteng now it would still run into this error if it is an IPv6 ip. You would have to change the OR to an AND operator.
You probably introduced that check in the first place as christoph also checked it, but the difference is that christoph needed this check to differentiate between IPv4 and IPv6 on aws which does not seem to be necessary for azure, hence I would suggest to remove it. `net.ParseCIDR(cidr)` should report any issues with invalid cidrs or ips

here the aws code from christoph:

```go
		if ip.To4() != nil {
			if permission.IpRanges == nil {
				permission.IpRanges = []*ec2.IpRange{}
			}

			permission.IpRanges = append(permission.IpRanges, &ec2.IpRange{
				CidrIp: &normalisedCIDR,
			})
		} else if ip.To16() != nil {
			if permission.Ipv6Ranges == nil {
				permission.Ipv6Ranges = []*ec2.Ipv6Range{}
			}

			permission.Ipv6Ranges = append(permission.Ipv6Ranges, &ec2.Ipv6Range{
				CidrIpv6: &normalisedCIDR,
			})
		}
```
https://github.com/gardener/gardener-extension-provider-aws/pull/318/files#diff-67e50731c2c35b38561d688bd0b216922c1f772bee2a8481630040bb8f70e4c3R219-R235

Reply:
fix and add test for ipv6 https://github.com/gardener/gardener-extension-provider-azure/pull/397/commits/1e295066b103affa709bd33ee58134b3df0197e5
Will check other places for IPv6 support.